src/solver/any/forward_any.cpp

```diff
@@ -0,0 +1,338 @@
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/kernel_info.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <miopen/target_properties.hpp>
+#include "miopen/buffer_info.hpp"
+#include "miopen/errors.hpp"
+#include "miopen/execution_context.hpp"
+#include "miopen/invoke_params.hpp"
+#include "miopen/tensor_view_utils.hpp"
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace any {
+
+constexpr uint64_t DivCeil(uint64_t numer, uint64_t denom) { return (numer + denom - 1) / denom; }
+
+MultiBufferWorkspaceTraits GetMultiBufferWorkspaceTraits(const TensorDescriptor& inputDesc)
+{
+    auto input_numel = inputDesc.GetElementSize();
+    auto size        = ((input_numel + LOCAL_SIZE - 1) / LOCAL_SIZE);
+
+    auto dtype = inputDesc.GetType();
+    size *= get_data_size(dtype);
+    size_t data_size      = get_data_size(dtype);
+    size_t workspace_size = AlignUp(size, LOCAL_SIZE) / LOCAL_SIZE;
+    size_t ws_scratch_mem = 2 * workspace_size * data_size;
+    size_t ws_local_mem   = LOCAL_SIZE * data_size;
+
+    return MultiBufferWorkspaceTraits{ws_scratch_mem, ws_local_mem};
+}
+
+bool AnyForward::IsApplicable(const ExecutionContext& context,
+                              const miopen::any::ProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    return true;
+}
+
+ConvSolution AnyForward::GetSolution(const ExecutionContext& context,
+                                     const miopen::any::ProblemDescription& problem) const
+{
+    std::ignore = context;
+
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto dtype        = problem.GetInputDesc().GetType();
+    auto input_dtype  = miopen::GetDataType(problem.GetInputDesc().GetType());
+    auto output_dtype = miopen::GetDataType(problem.GetOutputDesc().GetType());
+
+    std::cout << "input_dtype: " << input_dtype << std::endl;
+
+    auto input_dims  = problem.GetInputDesc().GetLengths();
+    auto output_dims = problem.GetOutputDesc().GetLengths();
+    auto dim         = problem.GetDim();
+
+    auto input_numel  = problem.GetInputDesc().GetElementSize();
+    auto output_numel = problem.GetOutputDesc().GetElementSize();
+
+    // auto in_dtype = input_dtype;
+    // if(in_dtype == "int8_t")
+    // {
+    //     in_dtype = "signed char";
+    // }
+    // else if(in_dtype == "bfloat16")
+    // {
+    //     in_dtype = "ushort";
+    // }
+
+    std::cout << "input_dtype: " << input_dtype << std::endl;
+
+    if(dim != -1)
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(output_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel        = KernelInfo{};
+        kernel.kernel_file = "MIOpenAny.cpp";
+        kernel.kernel_name = "AnyForward";
+
+        KernelBuildParameters build_params;
+
+        // Awkward way to handle different input types
```

Comment:
to solve this issue, inside the kernel, use the following lib: `#include <miopen_cstdint.hpp>` and pass INPUT_TYPE as `int8_t` as normal